voxblox/test/benchmark_e2e.cc

```diff
@@ -1,27 +1,84 @@
+#include <memory>
+
 #include <benchmark/benchmark.h>
 #include <benchmark_helpers/benchmark_helpers.h>
 
 #include "voxblox/core/tsdf_map.h"
 #include "voxblox/integrator/tsdf_integrator.h"
+#include "voxblox/integrator/tsdf_integrator_fast.h"
+#include "voxblox/simulation/sphere_simulator.h"
 
 using namespace voxblox;  // NOLINT
 
 class E2EBenchmark : public ::benchmark::Fixture {
  public:
-  void SetUp(const ::benchmark::State& st) {
-    // TODO(mfehr): IMPLEMENT
+  void SetUp(const ::benchmark::State& /*state*/) {
+    baseline_layer_.reset(new Layer<TsdfVoxel>(kVoxelSize, kVoxelsPerSide));
+    fast_layer_.reset(new Layer<TsdfVoxel>(kVoxelSize, kVoxelsPerSide));
+    baseline_integrator_.reset(
+        new TsdfIntegrator(config_, baseline_layer_.get()));
+    fast_integrator_.reset(
+        new fast::TsdfIntegrator(fast_config_, fast_layer_.get()));
+
+    sphere_points_C.clear();
+
+    constexpr double kMean = 0;
+    constexpr double kSigma = 0.05;
+    constexpr int kNumPoints = 100;
+    constexpr double kRadius = 1.0;
+
+    sphere_sim::createSphere(kMean, kSigma, kRadius, kNumPoints,
+                             &sphere_points_C);
+
+    colors.clear();
+    colors.resize(sphere_points_C.size(), Color(128, 255, 0));
+
+    T_G_C = Transformation();
```

Comment:
The main use case for voxblox is to integrate one frame at the time, i.e. one single camera origin/transformation. Therefore, I think it's ok to benchmark base one one camera origin. wdyt?